streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -752,6 +754,13 @@ public void shouldNotGetQueryMetadataWithPartitionerWhenNotRunningOrRebalancing(
         streams.queryMetadataForKey("store", "key", (topic, key, value, numPartitions) -> 0);
     }
 
+    @Test(expected = UnknownStateStoreException.class)
+    public void shouldThrowUnknownStateStoreExceptionWhenStoreNotExist() throws Exception {
+        final KafkaStreams streams = new KafkaStreams(getBuilderWithSource().build(), props, supplier, time);
```

Comment:
We should use a try-with-resources clause to make sure `close()` is called.

Seems, other tests in this class have a similar issue. Would be good to fix all test accordingly (if you want you can also to a separate PR for it).

Reply:
> We should use a try-with-resources clause to make sure `close()` is called.

Got it

> Seems, other tests in this class have a similar issue. Would be good to fix all test accordingly (if you want you can also to a separate PR for it).

Ok. I would like to create another PR for it.